src/clients/scopes/AddScopeDialog.tsx

```diff
@@ -22,24 +31,65 @@ import "./client-scopes.css";
 
 export type AddScopeDialogProps = {
   clientScopes: ClientScopeRepresentation[];
+  clientName?: string;
   open: boolean;
   toggleDialog: () => void;
   onAdd: (
-    scopes: { scope: ClientScopeRepresentation; type: ClientScopeType }[]
+    scopes: { scope: ClientScopeRepresentation; type?: ClientScopeType }[]
   ) => void;
+  isClientScopesConditionType?: boolean;
 };
 
+enum FilterType {
+  Name = "Name",
+  Protocol = "Protocol",
+}
+
+enum ProtocolType {
+  All = "All",
+  SAML = "SAML",
+  OpenIDConnect = "OpenID Connect",
+}
+
 export const AddScopeDialog = ({
   clientScopes,
+  clientName,
   open,
   toggleDialog,
   onAdd,
+  isClientScopesConditionType,
 }: AddScopeDialogProps) => {
   const { t } = useTranslation("clients");
   const [addToggle, setAddToggle] = useState(false);
   const [rows, setRows] = useState<ClientScopeRepresentation[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.Name);
+  const [protocolType, setProtocolType] = useState<ProtocolType>(
+    ProtocolType.All
+  );
```

Comment:
The type can be inferred since you are assigning a the enum value here , which means the generics can be omitted.
```suggestion
  const [filterType, setFilterType] = useState(FilterType.Name);
  const [protocolType, setProtocolType] = useState(ProtocolType.All);
```